### ISSUE TITLE
Investigate what it would mean to do benchmarks with Benchmark.NET in F# scripts

### ISSUE THREAD

**Issue by cartermp** [dotnet/fsharp#9519]
NuGet references let us pull in packages for scripts. Yay!

One such package is Benchmark.NET. It's a great library. However, it's not as straightforward as say, Json.NET. Given the following script:

```fsharp
#r "nuget:BenchmarkDotNet"

open BenchmarkDotNet.Attributes
open BenchmarkDotNet.Running

module Op =
    let inline not' x = if x then false else true
    let inline not'' x = match x with true -> false | false -> true


[<MemoryDiagnoser>]
type NotBench() =
    let s = "hello"

    [<Benchmark(Baseline=true)>]
    member _.Not() = 
        let mutable b = false
        for i=0 to 1 do b <- not (isNull s)
        b

    [<Benchmark>]
    member _.NotWithIf() = 
        let mutable b = false
        for i=0 to 1 do b <- Op.not' (isNull s)
        b

    [<Benchmark>]
    member _.NotWithMatch() = 
        let mutable b = false
        for i=0 to 1 do b <- Op.not'' (isNull s)
        b

let summary = BenchmarkRunner.Run<NotBench>()
printfn "%A" summary
```

You get this result when run via `dotnet fsi`:

<img width="983" alt="image" src="https://user-images.githubusercontent.com/6309070/85213245-8f1b4880-b310-11ea-8de5-bcbac8799ca5.png">

This is clearly thrown from Benchmark.NET, not F# Interactive. Is there something we could do to make this better? It's worth investigating IMO. Scripts are nicer for these kinds of things than console apps.

This may also have ramifications for doing similar things in notebooks.

**Comment by heronbpv** [comment]
This happens as well if you try to call Application.Run(new SomeForm()) in FSI. Something to do with loading dynamic assemblies in the FSI main loop, perhaps? Or starting another one while FSI's running. If the docs and (poor) directions I found about this exception are any indication, this may be by design (of the runtime). My tests where on .Net 4.8.
Bumped into it while exploring winforms in script environments.
Relevant links: 
[Stackoverflow about the exception](https://stackoverflow.com/questions/10091221/loading-plug-in-dll-files-the-invoked-member-is-not-supported-in-a-dynamic-ass)
[Docs, on the potential solution pointed above; didn't work for FSI for me](https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/loadfromremotesources-element?redirectedfrom=MSDN)

**Comment by abelbraaksma** [comment]
Indeed, I think FSI uses ambient assemblies and several BCL api's don't work well with that. But even if you would create concrete assemblies, let's say as a configuration option, I think that wouldn't solve the BDN issue, as it will take each benchmark, analysis the calling graph, and compiles a new (concrete) assembly. That assembly is then run, typically out-of-process, though you can opt-in to in-process.

**Comment by heronbpv** [comment]
Would it be possible to add this capability to the FSI API? Maybe with some way to either add the dlls or the "load from remote sources" app.config option.
E.g.: ```fsi.addDynamicDll path_to_dll```
